chainntnfs/interface.go

```diff
@@ -258,8 +258,11 @@ type ConfirmationEvent struct {
 // channels.
 func NewConfirmationEvent(numConfs uint32, cancel func()) *ConfirmationEvent {
 	return &ConfirmationEvent{
-		Confirmed:    make(chan *TxConfirmation, 1),
-		Updates:      make(chan uint32, numConfs),
+		// We cannot rely on the subscriber to immediately read from
+		// the channel so we need to create a larger buffer to avoid
+		// blocking the notifier.
+		Confirmed:    make(chan *TxConfirmation, numConfs),
```

Comment:
Why do we need a buffer of up to `numConfs`? Is this for some degenerate case where re-orgs constantly happen each time after a confirmation is notified?

Reply:
Turns out the `ntfn.Event.Confirmed` is fine as we only send out the confirmed event once, although the pattern is still dangerous as one single failure in sending to the `Confirmed` channel can lead to blocking other notifications.

The channel in question is the `ntfn.Event.Updates`, and as it's not filtered by the `dispatched` variable, we may send duplicate notifications there. To fix this, I added a new commit at the end to skip sending the duplicates.